File: leetcode/week_1_sprint/two_sum_hash_map_fundamentals/medium/two_sum_ii_input_array_is_sorted/Solution.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # Time Complexity O(n)
        # Space Complexity O(1)

        low: int = 0
        high: int = len(numbers) - 1
        ans: List[int] = [-1, -1]

        while (low < high):
            sum: int = numbers[low] + numbers[high]
            if (sum == target):
                ans[0] = low + 1
                ans[1] = high + 1
                break
            elif (sum < target):
                low += 1
            else:
                high -= 1
        
        return ans

File: leetcode/week_1_sprint/two_sum_hash_map_fundamentals/medium/two_sum_ii_input_array_is_sorted/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // Time Complexity O(n)
        // Space Complexity O(1)
        int low = 0;
        int high = numbers.size() - 1;
        vector<int> ans = {-1, -1};

        while (low < high) {
            int sum = numbers[low] + numbers[high];
            if (sum == target) {
                ans[0] = low + 1;
                ans[1] = high + 1;
                break;
            }

            else if (sum > target) {
                high--;
            }
            else {
                low++;
            }
        } 

        return ans;
    }
}